expression/builtin_string.go

```diff
@@ -1996,25 +1996,43 @@ type toBase64FunctionClass struct {
 }
 
 func (c *toBase64FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinToBase64Sig{newBaseBuiltinFunc(args, ctx)}
+	tp := types.NewFieldType(mysql.TypeVarString)
+	tp.Charset = charset.CharsetUTF8
+	tp.Collate = charset.CollationUTF8
+	tp.Flen = mysql.MaxBlobWidth
+
+	bf, err := newBaseBuiltinFuncWithTp(args, tp, ctx, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	sig := &builtinToBase64Sig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
 }
 
 type builtinToBase64Sig struct {
-	baseBuiltinFunc
+	baseStringBuiltinFunc
+}
+
+// neededEncodedLength return the base64 encoded string length
+func (b *builtinToBase64Sig) neededEncodedLength(n int) int {
+	length := (n + 2) / 3 * 4
+	return length + (length-1)/76
 }
 
-// eval evals a builtinToBase64Sig.
+// evalString evals a builtinToBase64Sig.
 // See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_to-base64
-func (b *builtinToBase64Sig) eval(row []types.Datum) (d types.Datum, err error) {
-	args, err := b.evalArgs(row)
-	if err != nil {
-		return d, errors.Trace(err)
+func (b *builtinToBase64Sig) evalString(row []types.Datum) (d string, isNull bool, err error) {
+	sc := b.ctx.GetSessionVars().StmtCtx
+	str, isNull, err := b.args[0].EvalString(row, sc)
+	if isNull || err != nil {
+		return "", isNull, errors.Trace(err)
 	}
-	str, err := args[0].ToString()
-	if err != nil {
-		return d, errors.Trace(err)
+
+	b.tp.Flen = b.neededEncodedLength(len(str))
```

Comment:
why we set flen during eval ?
we may use `neededEncodedLength(bf.args[0].Flen)` in getFunction() (compile phase)